instance/models/instance.py

```diff
@@ -119,11 +123,13 @@ class Instance(ValidateModelMixin, models.Model):
     # in a query, e.g. to do .select_related('ref_set')
     ref_set = GenericRelation(InstanceReference, content_type_field='instance_type', object_id_field='instance_id')
     openstack_region = models.CharField(
+        # TODO: Move this to OpenedXInstance
         max_length=16,
         blank=False,
         default=default_setting('OPENSTACK_REGION'),
     )
     tags = models.ManyToManyField(
+        # TODO: Move this to InstanceReference (common fields should be in non-abstract tables where possible)
```

Comment:
To me, it's a question of data modelling and normalization. If you put `tags` on the abstract model, and we introduce other `Instance` subclasses (e.g. "DaliteInstance"), then each subclass will also need to specify `tags = models.ManyToManyField` which is not DRY, and there will be additional many-to-many intermediate tables in the data model. So if we have additional Instance types and you wanted to query something like "how many Instances are tagged as `sandbox` vs `production`?", you would have to write some code to iterate over all the concrete `Instance` subclasses, and then to separately query the `tags` intermediate table for each one, and then combine the results.

On the other hand, if we put `tags` on the `InstanceReference`, then there is only ever one intermediate table created in the DB to implement the many-to-many relation, and it can be queried efficiently regardless of the number of Instance variations we may have in the future.

To achieve less typing and indirection in the django shell, you can just add an accessor to the `Instance` abstract class:

```python
@property
def tags(self):
    return self.ref.tags
```